gpt4all-chat/database.cpp

```diff
@@ -550,6 +550,7 @@ Database::Database(int chunkSize)
 
 Database::~Database()
 {
+    delete m_embLLM;
     m_dbThread.quit();
     m_dbThread.wait();
```

Comment:
There's a potential for a use-after-free if you delete m_embLLM while it is still being used by the database thread.
```suggestion
    m_dbThread.quit();
    m_dbThread.wait();
    delete m_embLLM;
```